documenters_aggregator/spiders/pbcc.py

```diff
@@ -0,0 +1,198 @@
+        """
+        event_link = item.css('a::attr(href)').extract_first()
+        if '?eID=' in event_link:
+            id_suffix = event_link.split('?eID=')[-1]
+        elif '?BID_ID=' in event_link:
+            id_suffix = event_link.split('?BID_ID=')[-1]
+        else:
+            raise ValueError('ID is required and not present in eID or BID_ID params')
+        return 'PBCC{}'.format(id_suffix)
+
+    def _parse_classification(self, item):
+        """
+        Parse or generate classification (e.g. town hall).
+
+        PBCC has relatively helpful classifications, expanding the AFB acronym.
+        """
+        classification = item.css('a b::text').extract_first().strip()
+        if classification == 'AFB':
+            return 'Advertisement for Bids'
+        else:
+            return classification
+
+    def _parse_status(self, item):
+        """
+        Parse or generate status of meeting. Can be one of:
+
+        * cancelled
+        * tentative
+        * confirmed
+        * passed
+
+        By default, return "tentative"
+        """
+        return 'tentative'
+
+    def _parse_location(self, item):
+        """
+        Parse or generate location. Url, latitutde and longitude are all
+        optional and may be more trouble than they're worth to collect.
+
+        TODO: Location details are inconsistent across event types, addresses
+        are mixed up with names and neither are reliably in the same place.
+        Could make manual exceptions for board meetings, or go to sub-page
+        """
+        return {
+            'url': None,
+            'name': None,
+            'coordinates': {
+                'latitude': None,
+                'longitude': None,
+            },
+        }
+
+    def _parse_all_day(self, item):
+        """
+        Parse or generate all-day status. Defaults to false.
+
+        Returns True if _parse_start_time is None
+        """
+        return self._parse_start_time(item) is None
+
+    def _parse_name(self, item, cal_date):
+        """
+        Parse or generate event name.
+        """
+        name_str = ''.join(item.xpath('./text()').extract()).rstrip().strip()
+        # Return without the beginning and end parentheses, additional spaces
+        # inside of them removed as well
+        clean_name_str = name_str[1:-1].strip()
+        if clean_name_str:
+            return '{} - {}'.format(self._parse_classification(item), clean_name_str)
+        else:
+            return '{} {}'.format(
+                self._parse_classification(item),
+                cal_date.strftime('%m/%d/%Y')
+            )
+
+    def _parse_description(self, item):
+        """
+        Parse or generate event name.
+
+        Other than the page-long info on AFBs, there isn't much of a description for
+        events outside of the classification and name, so returning None. Could be
+        implemented later for AFBs
+        """
+        return None
```

Comment:
If there is a [page like this](http://www.pbcchicago.com/content/working/opening_display.asp?BID_ID=502) it seems like it could be nice to stash in description.